src/context/AirDBContext.tsx

```diff
@@ -71,6 +81,129 @@ export const AirDBProvider = React.memo((props: any) => {
+          user.fields.password === password && user.fields.username === username
+      ).length > 0
+    const userId = (
+      users.filter(
+        (user: { fields: { username: string } }) =>
+          user.fields.username === username
+      )[0] || {
+        id: ''
+      }
+    ).id
+
+    if (isMatching) {
+      setToggled(false)
+      window.localStorage.setItem('username', username)
+      return updateAirDB('Testy - Users', userId, {
+        active: 'true'
+      })
+    } else {
+      throw new Error('INCORRECT PASSWORD')
+    }
+  }
+
+  const handleAddQuestion = ({ options, question }: any) => {
+    const userid = (
+      users.filter(
+        (user: { fields: { username: string } }) =>
+          user.fields.username === window.localStorage.getItem('username')
+      )[0] || { fields: { ID: '' } }
+    ).fields.ID
+
+    return postAirDB('Testy - Questions', {
+      ...question,
+      userid,
+      options: question.options
+        .sort()
+        .map((x: string) => x.trim())
+        .join(', '),
+      type: options.type,
+      autocheck: JSON.stringify(options.autocheck)
+    })
+  }
+
+  const handleAddTest = async ({ questionIDs, test }: any) => {
+    const userid = (
+      users.filter(
+        (user: { fields: { username: string } }) =>
+          user.fields.username === window.localStorage.getItem('username')
+      )[0] || { fields: { ID: '' } }
+    ).fields.ID
+
+    await postAirDB('Testy - Tests', {
+      ...test,
+      userid,
+      tags: test.tags.join(', '),
+      questions: questionIDs.join(', ')
+    }).then(() => (window.location.pathname = '/'))
+  }
+
+  const updateTestQuestions = ({ test, questionIDs }: any) => {
+    updateAirDB('Testy - Tests', test.id, {
+      questions: questionIDs.join(', ')
+    }).then(() => {
+      props.setRedirect('/')
+    })
+  }
+
+  const handleLogout = () => {
+    const userId = users.filter(
+      (user: any) =>
+        user.fields.username === window.localStorage.getItem('username')
+    )[0].id
+
+    return updateAirDB('Testy - Users', userId, {
+      active: false
+    })
+  }
+
+  const handleSetScore = ({ scoreID, score }: any) => {
+    const scorid = (
+      scores.filter(score => score.fields.ID === scoreID)[0] || {
+        id: ''
+      }
+    ).id
+
+    return updateAirDB('Testy - Test Scores', scorid, { score })
+  }
+
+  const createInitialScore = ({ id, setScoreID }: any) => {
+    const userid = (
+      users.filter(
+        user => user.fields.username === window.localStorage.getItem('username')
+      )[0] || { fields: { ID: 0 } }
+    ).fields.ID
+    postAirDB('Testy - Test Scores', {
+      userid,
+      test: id
+    }).then((r: any) => {
+      setScoreID(r[r.length - 1].fields.ID)
+    })
+  }
+
+  const handleQuestionAnswer = ({
+    handleSubmit,
+    value,
+    question,
+    index,
+    scoreID
+  }: any) => {
+    handleSubmit(value, question, index)
```

Comment:
You need to make sure you're returning the promises and then chaining them properly otherwise you have no control over when things happen asynchronously. For example, if you want this call and the subsequent one to run back to back you do this:
```
handleSubmit(value, question, index).then(() => ...blah...)
```

If you want them to happen simultaneously, do this:
```
Promise.all([
  handleSubmit(value, question, index),
  postAirDB('Testy - Test Instances'... )
])
```

Regardless, you need to make sure you have the good practice of returning the results of your asynchronous calls (i.e. the promise result). That way, the caller has a chance to do something with the result or at least do something after it's done (or even concurrently).